src/lib/accessibility/FocusZone/FocusTrapZone.tsx

```diff
@@ -0,0 +1,296 @@
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
+import * as keyboardKey from 'keyboard-key'
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+import eventStack from '../../eventStack'
+
+import {
+  getNextElement,
+  getFirstTabbable,
+  getLastTabbable,
+  getWindow,
+  focusAsync,
+  HIDDEN_FROM_ACC_TREE,
+} from './focusUtilities'
+
+import { FocusTrapZoneProps } from './FocusTrapZone.types'
+import getUnhandledProps from '../../getUnhandledProps'
+import getElementType from '../../getElementType'
+import * as customPropTypes from '../../customPropTypes'
+
+export class FocusTrapZone extends React.Component<FocusTrapZoneProps, {}> {
+  private static _focusStack: FocusTrapZone[] = []
+  private _root: { current: HTMLElement | null } = { current: null }
+  private _previouslyFocusedElementOutsideTrapZone: HTMLElement
+  private _previouslyFocusedElementInTrapZone?: HTMLElement
+  private windowElement: Window | null
+
+  private createRef = elem => (this._root.current = ReactDOM.findDOMNode(elem) as HTMLElement)
+
+  static propTypes = {
+    as: customPropTypes.as,
```

Comment:
wouldn't we provide any descriptions for these?

Reply:
Description for props is provided here ```FocusTrapZone.types.tsx```. Do you think we should duplicate it?